compiler/rustc_lexer/src/cursor.rs

```diff
@@ -59,6 +59,15 @@ impl<'a> Cursor<'a> {
         iter.next().unwrap_or(EOF_CHAR)
     }
 
+    /// Peeks the third symbol from the input stream without consuming it.
+    pub fn third(&self) -> char {
```

Comment:
Since `second` and `third` are exclusively used in the error path if I'm not mistaken (haven't double-checked), perf shouldn't matter that much and we can maybe avoid introducing those helpers? Idk, do we have access to `self.chars` in the parser? If so, we could just use `.clone().nth(N)`, right?

Reply:
`self.chars` is private to the lexer. I initially made it public, but given it is used only in one place it felt better to introduce the method. But then we could instead just have an `nth` method that expands to `.clone().nth(N)` instead.